testing_and_setup/compass/manage_regression_suite.py

```diff
@@ -193,19 +194,254 @@ def process_test_clean(test_tag, work_dir):  # {{{
 
     print("   -- Cleaned case '{}': -o {} -c {} -r {} -t {}".format(
         test_name, test_core, test_configuration, test_resolution, test_test))
 
     dev_null.close()
+    # }}}
 
-# }}}
+
+def local_setup_script(
+        work_dir, suite_tag, verbose, nodes , suite_root, testcase_data, model_runtime, config_file, baseline_dir, isparallel):
+    # {{{
+    try:
+        suite_name = suite_tag.attrib['name']
+    except KeyError:
+        print("ERROR: <regression_suite> tag is missing 'name' attribute.")
+        print('Exiting...')
+        sys.exit(1)
+
+    if not os.path.exists('{}'.format(work_dir)):
+        os.makedirs('{}'.format(work_dir))
+
+    if verbose:
+        # flush existing regression suite output file
+        open(work_dir + '/manage_regression_suite.py.out', 'w').close()
+
+    # creates the script writer and generates the top section of the code
+    if isparallel: 
+        script_name = '{}/parallel_{}.py'.format(work_dir, suite_name)
+        local_script = open('{}'.format(script_name), 'w')
+        local_code = write_local_parallel_top(
+            work_dir, suite_tag, nodes)
+    else:
+        script_name = '{}/{}.py'.format(work_dir, suite_name)
+        local_script = open('{}'.format(script_name), 'w')
+        local_code = write_script_top(
+            work_dir)
+
+    # makes the script executable
+    dev_null = open('/dev/null', 'a')
+    subprocess.check_call(['chmod',
+                           'a+x',
+                           '{}'.format(script_name)],
+                           stdout=dev_null,
+                           stderr=dev_null)
+    dev_null.close()
+
+    # sets up rest of the code 
+    local_code = setup_suite(
+        suite_root,
+        work_dir,
+        model_runtime,
+        config_file,
+        baseline_dir,
+        verbose, isparallel, local_code, testcase_data)
+    
+
+    return local_script, local_code
+    # }}}
+
+
+def write_local_parallel_data(local_parallel_code, work_dir, testcase_data):
+        # {{{
+    local_parallel_code += "testcase_data = {}\n"
+    for name in testcase_data.keys():
+        local_parallel_code += "testcase_data['{}'] = {}\n".format(name, testcase_data[name])
+    return local_parallel_code
+    # }}}
+
+
+def write_local_parallel_bottom(local_parallel_code):
+    #{{{ 
+    local_parallel_code += "# rewrite algorithm to read in testcase_data\n# must use args.work_dir instead of os.getcwd\n"
+    local_parallel_code += "def run(key):\n"
+    local_parallel_code += "\tos.chdir(base_path)\n"
+    local_parallel_code += "\tcase_output = open('case_outputs/{}'.format(key.replace(' ', '_')), 'w+') \n"
+    local_parallel_code += "\tprint('\t\t** Running case {} **'.format(key))\n"
+    local_parallel_code += "\tos.chdir('./{}'.format(testcase_data[key]['path']))\n"
+    local_parallel_code += "\tcommand = ['time', '-p', './{}'.format(testcase_data[key]['filename'])]\n"
+    local_parallel_code += "\tprint('{}'.format(command))\n"
```

Comment:
```suggestion
    local_parallel_code += "\tprint('{}'.format(' '.join(command)))\n"
```
If you want to keep this at all, it should be formatted just a s a command, not as a list of strings.  I don't think it's useful, though, and I would just remove this line.